src/calendar/week.tsx

```diff
@@ -1,155 +1,98 @@
 import eachDay from 'date-fns/each_day'
-import endOfWeek from 'date-fns/end_of_week'
 import format from 'date-fns/format'
-import isAfter from 'date-fns/is_after'
-import isBefore from 'date-fns/is_before'
-import isEqual from 'date-fns/is_equal'
-import isSameDay from 'date-fns/is_same_day'
-import isSameMonth from 'date-fns/is_same_month'
 import isWeekend from 'date-fns/is_weekend'
-import isWithinRange from 'date-fns/is_within_range'
-import startOfDay from 'date-fns/start_of_day'
-import startOfWeek from 'date-fns/start_of_week'
-import React, { Component, SyntheticEvent } from 'react'
+import React, { FC, SyntheticEvent } from 'react'
 
-import { IDate, IDayRenderProps, IWeekRenderProps } from '../@types'
+import { IDayRenderProps, IWeekRenderProps } from '../@types'
 import Day from './day'
 
 export type Props = {
-  activeMonth: IDate
   blockClassName: string
   customRender?: IWeekRenderProps
-  date: IDate
-  disabledIntervals?: {
-    start: IDate
-    end: IDate
-  }[]
-  highlightedEnd?: IDate
-  highlightedStart?: IDate
-  maxDate?: IDate
-  minDate?: IDate
-  onDayClick: (event: SyntheticEvent<HTMLButtonElement>) => void
-  onDayMouseEnter: (event: SyntheticEvent<HTMLButtonElement>) => void
-  onDisabledDayClick: (event: SyntheticEvent<HTMLButtonElement>) => void
+  getDayIsCurrentMonth: (date: Date) => boolean
+  getDayIsDisabled: (date: Date) => boolean
+  getDayIsHighlighted: (date: Date) => boolean
+  getDayIsNextMonth: (date: Date) => boolean
+  getDayIsSelectable: (date: Date) => boolean
+  getDayIsSelected: (date: Date) => boolean
+  getDayIsSelectedMax: (date: Date) => boolean
+  getDayIsSelectedMin: (date: Date) => boolean
+  getDayIsToday: (date: Date) => boolean
+  getDayIsWeekend: (date: Date) => boolean
+  handleOnDayClick: (event: SyntheticEvent<HTMLButtonElement>) => void
+  handleOnDayMouseEnter: (event: SyntheticEvent<HTMLButtonElement>) => void
+  handleOnDisabledDayClick: (event: SyntheticEvent<HTMLButtonElement>) => void
   renderDay?: IDayRenderProps
-  selectedMax?: IDate
-  selectedMin?: IDate
-  today: IDate
-  weekStartsOn: number
+  weekEnd: Date
+  weekNumber: number
+  weekStart: Date
 }
 
-export default class Week extends Component<Props, {}> {
-  _dateSelectable(date: Date) {
-    const {minDate, maxDate} = this.props
-
-    if (this._dateDisabled(date)) {
-      return false
-    }
-
-    if (minDate && maxDate) {
-      return isWithinRange(date, minDate, maxDate)
-    } else if (minDate && !maxDate) {
-      return isAfter(date, minDate) || isEqual(date, minDate)
-    } else if (maxDate && !minDate) {
-      return isBefore(date, maxDate) || isEqual(date, maxDate)
-    } else {
-      return true
-    }
-  }
-
-  _dateSelected(date: Date) {
-    const {selectedMin, selectedMax} = this.props
-    return Boolean(
-      selectedMin && selectedMax && isWithinRange(startOfDay(date), startOfDay(selectedMin), startOfDay(selectedMax))
+const Week: FC<Props> = (props) => {
```

Comment:
tbh, I don't see the reason to use `FC`, there is no even `children` here

Reply:
Can you explain it, FC as a type declaration holds up some minor type helpers as well, optional props handling, return type definition. Or probably I see it wrong?